src/tests/unit/electron/device-connect-view/components/device-connect-body.test.tsx

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { BrowserWindow } from 'electron';
+import { shallow } from 'enzyme';
+import * as React from 'react';
+import {
+    DeviceConnectBody,
+    DeviceConnectBodyProps,
+    DeviceConnectBodyState,
+} from '../../../../../electron/device-connect-view/components/device-connect-body';
+import { DeviceConnectPortEntryProps } from '../../../../../electron/device-connect-view/components/device-connect-port-entry';
+
+describe('DeviceConnectBodyTest', () => {
+    const props: DeviceConnectBodyProps = {
+        currentWindow: {
+            close: () => {
+                return;
+            },
+        } as BrowserWindow,
+    };
+
+    const expectedBeforeState: DeviceConnectBodyState = {
+        canStartTesting: false,
+        needsValidation: true,
+        isConnecting: false,
+    };
+
+    const expectedStateWhileConnecting: DeviceConnectBodyState = {
+        isConnecting: true,
+        canStartTesting: false,
+        needsValidation: true,
+        connectedDevice: '',
+    };
+
+    const expectedStateConnectedSuccess: DeviceConnectBodyState = {
+        isConnecting: false,
+        canStartTesting: true,
+        needsValidation: false,
+        connectedDevice: 'Test Device Name!',
+    };
+
+    const expectedStateConnectedFail: DeviceConnectBodyState = {
+        isConnecting: false,
+        canStartTesting: false,
+        needsValidation: true,
+        connectedDevice: '',
+    };
+
+    test('render', () => {
+        const rendered = shallow(<DeviceConnectBody {...props} />);
+        rendered.childAt(1).props();
+        expect(rendered.getElement()).toMatchSnapshot();
+    });
+
+    test('OnConnectingCallback updates state', () => {
```

Comment:
You can use `test.each` to refactor this test down; but that doesn't necessarily have to be this PR.
Just a thought.

Reply:
Looking a little closer, I'm actually calling different functions in some of these state related tests, so I'm not sure `test.each` would help things out. I will look into it more for a future PR.